plugins/operators/__init__.py

```diff
@@ -1416,6 +1416,202 @@ def execute(self, ctx):
         ctx.dataset.create_index(field_name, unique=unique, wait=False)
 
 
+class ConvertIndexToUnique(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="convert_index_to_unique",  # Must match what's in fiftyone.yml
+            label="Convert an existing index to unique",
+            description="Convert an existing non-unique index to a unique index in the dataset if possible or identify any unique key violations that would occur.",
+            allow_immediate_execution=True,
+            allow_delegated_execution=True,
+            default_choice_to_delegated=False,
+        )
+
+    def resolve_input(self, ctx):
+        import re
+
+        inputs = types.Object()
+        indexes = ctx.dataset.get_index_information()
+
+        if not indexes:
+            inputs.str(
+                "index_name",
+                label="This dataset has no non-unique indexes to update",
+                view=types.Warning(),
+            )
+            return types.Property(
+                inputs,
+                invalid=True,
+            )
+
+        indexes = [
+            i
+            for i, v in indexes.items()
+            if not re.match(r"\.?id$", i) and "unique" not in v
+        ]
+
+        index_selector = types.DropdownView()
+        for key in indexes:
+            index_selector.add_choice(key, label=key)
+
+        inputs.str(
+            "index_name",
+            label="Index",
+            description="The existing index to make unique",
+            view=index_selector,
+            required=True,
+        )
+        inputs.bool(
+            "dry_run",
+            required=False,
+            default=False,
+            label="Check for unique key violations only (dry run)",
+        )
+
+        return types.Property(
+            inputs,
+            view=types.View(label="Convert existing index to unique"),
+        )
+
+    def execute(self, ctx):
+        field = ctx.params.get("index_name")
```

Comment:
`field` is a potentially confusing variable name as index names are only equal to field names in certain cases. Compound indexes and frame indexes are cases where `index_name != field_name`. I'd recommend calling this variable `name` or `index_name` for clarity